scripts/custom_lint_checks.py

```diff
@@ -73,17 +74,77 @@ def visit_call(self, node):
         Args:
             node. Call. The current function call node.
         """
-        if (isinstance(node.func, astroid.Name) and
+        if (isinstance(node.func, astroid.Name)) and (
                 node.func.name == self._function_name):
-            if (node.args is not None and
+            if (node.args is not None) and (
                     len(node.args) > self._positional_arguments_count):
                 self.add_message('non-explicit-kwargs', node=node)
 
 
+class HangingIndentChecker(BaseChecker):
+    """Custom pylint checker which checks for break after paren in case
+    of hanging indentation.
+    """
+    __implements__ = IRawChecker
+
+    name = 'hanging-indent'
+    priority = -1
+    msgs = {
+        'C0002': (
+            (
+                'There should be a break after parenthesis when content within '
+                'paren spans to multiple lines.'),
+            'break-after-hanging-indent',
+            (
+                'If something within parenthesis extends along multiple lines, '
+                'break after opening paren.')
+        ),
+    }
+
+    def process_module(self, node):
+        """Process a module.
+
+        Args:
+            node: Node to access module content.
+        """
+        file_content = node.stream().readlines()
+        file_length = len(file_content)
+        exclude = False
+        for line_num in range(file_length):
+            line = file_content[line_num].lstrip().rstrip()
+            if line.startswith('"""') and not line.endswith('"""'):
+                exclude = True
+            if line.endswith('"""'):
+                exclude = False
+            if line.startswith('#') or exclude or '# pylint: disable' in line:
+                continue
+            line_length = len(line)
+            bracket_cnt = 0
+            for char_num in range(line_length):
+                char = line[char_num]
+                if char == '(': # pylint: disable=hanging-indent
+                    if bracket_cnt == 0:
+                        pos = char_num
+                    bracket_cnt += 1
+                elif char == ')' and bracket_cnt > 0:
+                    bracket_cnt -= 1
+            if bracket_cnt > 0 and pos + 1 < line_length:
+                content = line[pos + 1:]
+                if not len(content) or not ',' in content:
+                    continue
+                split_list = content.split(', ')
+                punct = set('@^! #%$&)(+*-=')
```

Comment:
Can you please explain the logic behind choosing this set? Also, I don't think these are exactly punctuations so it would be better if we go for a more descriptive name like operators or special characters (something similar maybe).

Reply:
Hi @apb7, since I wanted a list which used only commas for separation, I included all other possible separators in this list. On a check now, I find that some of the cases are skipped. I think we could get better results if we use a regex here. What do you suggest?

Reply:
I'm not sure about the regex approach. Can you please check the time taken by both the approaches and then chose the more efficient one?

Reply:
Sure @apb7, I will make the changes and update you, Thanks!

Reply:
I have checked both the approaches and they produced almost similar error messages. The current approach is more efficient in terms of time taken and we can go with it.

Reply:
Okay, thanks!